actors/miner/src/lib.rs

```diff
@@ -1209,7 +1191,8 @@ impl Actor {
                 update: usi.update,
                 sector_info: &usi.sector_info,
                 deal_info: deal_activation,
-                full_unsealed_cid: computed_commd,
+                full_unsealed_cid: CompactCommD::new(usi.update.new_unsealed_cid)
+                    .get_cid(usi.sector_info.seal_proof)?,
```

Comment:
This change is incorrect, related to the comment above. The computed CommD is the only one we know is present. If one was declared, we can check it, but it may not have been. The result here can never be empty (today), so `.get_cid(...)` signals a problem.

Reply:
I dived in to try to hit this in a test. The case that I think will go wrong is when PRU1 is used, no CommD specified, but there are deals. That will lead to zero CID here. This will be passed to proof verification and should fail to match the CommR.

We can't test this today because the test VM always succeeds in `verify_replica_update`. We don't have tracing or dependency injection that would let us intercept this, though @alexytsu has been touching on the `Primitives` (for which we do have fakes) in unifying our test VMs. Injecting the `Verifier` is probably the way to go.